google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/MutateRowOptions.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigtable.data.v2.models;
+
+import com.google.common.base.Preconditions;
+import java.io.Serializable;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * Represents a collection of customized options for mutating a row. Currently, it only wraps an
+ * authorized view id option but there could be more in the future.
+ */
+public class MutateRowOptions implements Serializable {
```

Comment:
You can use AutoValue for all the Options class. Something like:

```
@AutoValue
public abstract class Options {

    @Nullable private String viewId;

    public static Options create() {
        return new AutoValue_Options(null);
    }

    public static Options createForView(String view) {
        return new AutoValue_Options(view);
    }

    @Nullable abstract String getViewId();
}
```

Reply:
Somehow the code doesn't compile for me.

My code:

```
@AutoValue
public abstract class ExistsOptions implements Serializable {
  @Nullable private String authorizedViewId;

  public static ExistsOptions create() {
    return new AutoValue_ExistsOptions(null);
  }

  public static ExistsOptions createForAuthorizedView(String authorizedViewId) {
    return new AutoValue_ExistsOptions(authorizedViewId);
  }

  @Nullable
  public String getAuthorizedViewId() {
    return this.authorizedViewId;
  }
}
```
And the error was `AutoValue_ExistsOptions() Expected 0 arguments but found 1`.